lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -5925,6 +5927,12 @@ void TypeChecker::checkConformancesInContext(DeclContext *dc,
                    emitWarning ? diag::nscoding_unstable_mangled_name_warn
                                : diag::nscoding_unstable_mangled_name,
                    *kind, classDecl->TypeDecl::getDeclaredInterfaceType());
+          if (isFixable) {
+            diagnose(classDecl, diag::add_nskeyedarchivelegacy_attr)
+              .fixItInsert(
+                classDecl->getAttributeInsertionLoc(/*forModifier=*/false),
+                "@NSKeyedArchiveLegacy(\"<#class archival name#>\")");
```

Comment:
We need to actually provide the old mangled name here, rather than trust users to get it right.